src/main/java/com/vint/shop/domain/User.java

```diff
@@ -1,68 +1,92 @@
 package com.vint.shop.domain;
 
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
 import javax.persistence.*;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
 import java.time.LocalDate;
+import java.util.Collection;
 import java.util.Objects;
+import java.util.Set;
 
 @Entity
 @Table(name = "\"user\"")
-public class User { //изменить название класса
+public class User implements UserDetails {
```

Comment:
It's not quite a good idea to implement such interface, as you have to implement some methods< which are not related to the DB table